ocaml/typing/jkind.ml

```diff
@@ -717,6 +837,29 @@ module Const = struct
       List.map parse_mode modes
   end
 
+  let jkind_of_product_annotations jkinds =
+    let folder (layouts, mode_ub, ext_ub, null_ub)
+        { layout;
+          modes_upper_bounds;
+          externality_upper_bound;
+          nullability_upper_bound
+        } =
+      ( layout :: layouts,
+        Modes.join mode_ub modes_upper_bounds,
+        Externality.join ext_ub externality_upper_bound,
+        Nullability.join null_ub nullability_upper_bound )
+    in
+    let layouts, mode_ub, ext_ub, null_ub =
+      List.fold_left folder
+        ([], Modes.min, Externality.min, Nullability.max)
```

Comment:
```suggestion
        ([], Modes.min, Externality.min, Nullability.min)
```

?

Probably worth adding some tests around this function.... though I suppose the existing mode-crossing tests are the tests. Externality and nullability don't matter for non-values.

Reply:
Well spotted - a hold-over from the earlier discussion of whether we should take a join or meet of nullability.  Fixed, and a test added.